### ISSUE TITLE
license of YOLOv8 and cuda

### ISSUE THREAD

**Issue by hiDevman** [ultralytics/ultralytics#8332]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

YOLOv8 is under AGPL and cuda is NVIDA License.
How can YOLOv8 include cuda?
cuda is not an open source software, so it cannot be disclosed.
I wonder relationship between YOLOv8 and cuda.

**Comment by hiDevman** [comment]
OK, Then Can I use YOLO with CUDA Toolkit, cuDNN?
Do I need to disclose the source code of CUDA Toolkit, cuDNN?

**Comment by glenn-jocher** [comment]
@hiDevman absolutely, you can use YOLO with the CUDA Toolkit and cuDNN to leverage GPU acceleration for improved performance. There's no need to disclose the source code of CUDA Toolkit or cuDNN, as they are used as libraries and tools. Your usage complies with their licenses when you use them as intended for developing and deploying your applications. 😊🚀

**Comment by hiDevman** [comment]
YOLO uses torch and torch uses cuda toolkit and cuDnn.
I understand that it is no problem as you mentioned. Is it right?

**Comment by glenn-jocher** [comment]
@hiDevman That's correct! YOLO utilizes PyTorch, which in turn uses the CUDA Toolkit and cuDNN for GPU acceleration. This usage is perfectly fine and doesn't require you to disclose the source code of CUDA Toolkit or cuDNN. Happy coding! 😊👍
